controllers/oneagent/daemonset/affinity.go

```diff
@@ -0,0 +1,36 @@
+package daemonset
+
+import corev1 "k8s.io/api/core/v1"
+
+func (dsInfo *builderInfo) affinity() *corev1.Affinity {
+	return &corev1.Affinity{
+		NodeAffinity: &corev1.NodeAffinity{
+			RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+				NodeSelectorTerms: dsInfo.affinityNodeSelectorTerms(),
+			},
+		},
+	}
+}
+
+func (dsInfo *builderInfo) affinityNodeSelectorTerms() []corev1.NodeSelectorTerm {
+	return []corev1.NodeSelectorTerm{
+		kubernetesArchOsSelectorTerm(),
+	}
+}
+
+func kubernetesArchOsSelectorTerm() corev1.NodeSelectorTerm {
+	return corev1.NodeSelectorTerm{
+		MatchExpressions: []corev1.NodeSelectorRequirement{
+			{
+				Key:      "kubernetes.io/arch",
```

Comment:
We have constants for these in `activegate/reconciler/statefulset/statefulset.go` so we could reuse these in some way.

Can we remove the beta affinity for that `statefulset` as well?

Reply:
removed from other deployments and refactored